File: ros_ws/src/intro_pkg1/scripts/NumberPublisher.py
#!/usr/bin/env python
import rospy
# Message class
from std_msgs.msg import Float64

# Main function
def main():
    # Initialize node
    rospy.init_node('NumberPublisher', anonymous= True)
    # Publisher object
    publishingObj = rospy.Publisher('floating_numbers', Float64, queue_size=10)
    # Rate controller. 2 time a second, 2 Hz
    rateHandler = rospy.Rate(2)
    # Variable that we'll publish
    a = 0.0
    # Main loop while ROS is running
    while(not rospy.is_shutdown()):
        # Make a message
        msg = Float64()
        msg.data = a
        # Update the variable that we'll send
        # Could also do: a = 0 if a > 10 else a + 0.5
        a += 0.5
        if a > 10.0:
            a = 0.0
        # Publish the message
        publishingObj.publish(msg)
        # Hold execution using the rate handler
        rateHandler.sleep()


# If node run directly
if __name__ == "__main__":
    # Main try ... except ... case
    try:
        main()
    except rospy.ROSInterruptException:
        rospy.logfatal("Node interrupted from operating, terminating operations")

File: ros_ws/src/src/intro_pkg1/src/NumberPublisher.cpp
// Header file for ROS
#include "ros/ros.h"
// Header file for messages
#include "std_msgs/Float64.h"

int main(int argc, char **argv)
{
    // Initialize the node, name it NumberPublisher
    ros::init(argc, argv, "NumberPublisher");
    // Create a node handler
    ros::NodeHandle nodeHandler;
    // Create a publisher object. Topic name: "/floating_numbers", queue_size: 10
    ros::Publisher publishingObj = nodeHandler.advertise<std_msgs::Float64>("floating_numbers", 2);
    // Rate handler. 2 times a second = 2Hz
    ros::Rate rateHandler = ros::Rate(2);

    // Floating point numbers
    double number = 0.0;
    // While loop to publish messages
    while(ros::ok()) {
        // Message
        std_msgs::Float64 pubMsg;
        // Floating number
        if (number > 10) {
            number = 0.0;
        }
        pubMsg.data = number;  // Assign the number to the message to be published
        number += 0.5;
        // Publish the message
        publishingObj.publish(pubMsg);

        // Delay for satisfying the rate
        rateHandler.sleep();
    }

    return 0;
}
